### PR TITLE
TSDK-317 | Added validation rules to Genus

### PR DESCRIPTION
**Pull request by ignaciofernandezsoto**
## Purpose
Validation rules such as optional and non optional fields are added

## Approach
Set the corresponding fields as optional on Genus protobuf files

## Testing
Protobuf compiler runs correctly and the project structure looks correct.

## Tickets
* TSDK-317

### CODE REVIEW

File: proto/genus/genus_models.proto [hunk 1]
```diff
@@ -90,10 +93,10 @@ message IndexSpec {
   // The name of the index
   string indexName = 1;
   // A description of the fields to be indexed
-  IndexFieldSpec indexFieldSpec = 3;
+  IndexFieldSpec indexFieldSpec = 3 [(validate.rules).message.required = true];
```
Comment (nandotorterolo) [inline proto/genus/genus_models.proto:96]: I am not sure if it was an intentional decision, but field index number 2 is missing. (and is not related to this PR)

**Review by nandotorterolo — APPROVED** [review]

File: proto/genus/genus_models.proto [hunk 2]
```diff
@@ -24,10 +27,10 @@ enum TxoState {
 
 // A Box and its status
 message Txo {
-  brambl.models.box.Box box = 1;
+  brambl.models.box.Box box = 1 [(validate.rules).message.required = true];
   TxoState state = 2;
-  brambl.models.KnownIdentifier.TransactionOutput32 id = 3;
-  brambl.models.Address address = 4;
+  brambl.models.KnownIdentifier.TransactionOutput32 id = 3 [(validate.rules).message.required = true];
+  brambl.models.Address address = 4 [(validate.rules).message.required = true];
```
Comment (nandotorterolo) [inline proto/genus/genus_models.proto:33]: in the current Txo Bifrost Node I can see these differences.
- state is required
- address is optional

File: proto/genus/genus_models.proto (hunk 1 above)
Comment (ignaciofernandezsoto) [reply proto/genus/genus_models.proto:96]: Good catch! Fixed it 😸

File: proto/genus/genus_models.proto (hunk 2 above)
Comment (ignaciofernandezsoto) [reply proto/genus/genus_models.proto:33]: Good catch!

**Review by DiademShoukralla — APPROVED** [review]
LGTM!

**Review by SeanCheatham — APPROVED** [review]

**Pull request merged**